src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -823,23 +856,23 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
                             operationBlockScope, operationBlocks, declaredSymbol, semanticModel.Compilation, AnalyzerOptions,
                             GetControlFlowGraph, declaredNode.SyntaxTree, filterSpan, isGeneratedCode, cancellationToken);
 
-                        // Catch Exception from the start action.
                         ExecuteAndCatchIfThrows(
                             startAction.Analyzer,
-                            data =>
-                            {
-                                data.action(data.context);
-                                data.blockEndActions?.AddAll(data.scope.OperationBlockEndActions);
-                                data.operationActions?.AddRange(data.scope.OperationActions);
-                            },
-                            (action: operationBlockStartAction.Action, context: operationStartContext, scope: operationBlockScope, blockEndActions: operationBlockEndActions, operationActions: operationActions),
+                            static data => data.operationBlockStartAction.Action(data.operationStartContext),
+                            (operationBlockStartAction, operationStartContext),
                             new AnalysisContextInfo(Compilation, declaredSymbol),
                             cancellationToken);
+
+                        operationBlockEndActions!.AddAll(operationBlockScope.OperationBlockEndActions);
```

Comment:
If this really is always non-null, it would be better to change its initial assignment from try-cast to direct cast.

Reply:
This code gets rewritten in a follow-up where there is no cast at all. This pr is just about pulling out mechanical, simple, changes from the more complex pr.